ethapi/sfc_api.go

```diff
@@ -0,0 +1,304 @@
+	v, err := s.b.GetStakerDelegatorsClaimedRewards(ctx, idx.StakerID(stakerID))
+	if err != nil {
+		return nil, err
+	}
+	return (*hexutil.Big)(v), err
+}
+
+// RPCMarshalStaker converts the given staker to the RPC output .
+func RPCMarshalStaker(it sfctype.SfcStakerAndID) map[string]interface{} {
+	return map[string]interface{}{
+		"id":               hexutil.Uint64(it.StakerID),
+		"totalStake":       (*hexutil.Big)(it.Staker.CalcTotalStake()),
+		"stake":            (*hexutil.Big)(it.Staker.StakeAmount),
+		"delegatedMe":      (*hexutil.Big)(it.Staker.DelegatedMe),
+		"isValidator":      it.Staker.IsValidator,
+		"isCheater":        it.Staker.IsCheater,
+		"address":          it.Staker.Address,
+		"createdEpoch":     hexutil.Uint64(it.Staker.CreatedEpoch),
+		"createdTime":      hexutil.Uint64(it.Staker.CreatedTime),
+		"deactivatedEpoch": hexutil.Uint64(it.Staker.DeactivatedEpoch),
+		"deactivatedTime":  hexutil.Uint64(it.Staker.DeactivatedTime),
+	}
+}
+
+func (s *PublicSfcAPI) addStakerMetricFields(ctx context.Context, res map[string]interface{}, stakerID idx.StakerID) (map[string]interface{}, error) {
+	blocks, period, err := s.b.GetDowntime(ctx, idx.StakerID(stakerID))
+	if err != nil {
+		return nil, err
+	}
+	res["missedBlocks"] = hexutil.Uint64(blocks)
+	res["downtime"] = hexutil.Uint64(period)
+
+	poi, err := s.b.GetStakerPoI(ctx, stakerID)
+	if err != nil {
+		return nil, err
+	}
+	res["poi"] = (*hexutil.Big)(poi)
+
+	validatingPower, err := s.b.GetValidatingPower(ctx, stakerID)
+	if err != nil {
+		return nil, err
+	}
+	res["validatingPower"] = (*hexutil.Big)(validatingPower)
+
+	validationScore, err := s.b.GetValidationScore(ctx, stakerID)
+	if err != nil {
+		return nil, err
+	}
+	res["validationScore"] = (*hexutil.Big)(validationScore)
+
+	originationScore, err := s.b.GetOriginationScore(ctx, stakerID)
+	if err != nil {
+		return nil, err
+	}
+	res["originationScore"] = (*hexutil.Big)(originationScore)
+
+	claimedRewards, err := s.b.GetStakerClaimedRewards(ctx, stakerID)
+	if err != nil {
+		return nil, err
+	}
+	res["claimedRewards"] = (*hexutil.Big)(claimedRewards)
+
+	delegatorsClaimedRewards, err := s.b.GetStakerDelegatorsClaimedRewards(ctx, stakerID)
+	if err != nil {
+		return nil, err
+	}
+	res["delegatorsClaimedRewards"] = (*hexutil.Big)(delegatorsClaimedRewards)
+
+	return res, nil
+}
+
+func (s *PublicSfcAPI) addDelegatorMetricFields(ctx context.Context, res map[string]interface{}, address common.Address) (map[string]interface{}, error) {
+	claimedRewards, err := s.b.GetDelegatorClaimedRewards(ctx, address)
+	if err != nil {
+		return nil, err
+	}
+	res["claimedRewards"] = (*hexutil.Big)(claimedRewards)
+
+	return res, nil
+}
+
+// GetStaker returns SFC staker's info
+func (s *PublicSfcAPI) GetStaker(ctx context.Context, stakerID hexutil.Uint, verbosity hexutil.Uint64) (map[string]interface{}, error) {
```

Comment:
add comment about verbosity can affect return result.

Do these methods with verbosity parameter belong to the public API?
Can't see them defined in lines 88-100 of PublicSfcAPI.

Reply:
Yes, verbosity parameters belong to the public API. Methods on 88-100 don't need verbosity because they return specific values, not containers